Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/LibraryManager.java

```diff
@@ -325,7 +326,17 @@ private TranslatedLibrary generateTranslatedLibrary(Library library) {
             }
             if (library.getStatements() != null && library.getStatements().getDef() != null) {
                 for (ExpressionDef expressionDef : library.getStatements().getDef()) {
-                    translatedLibrary.add(expressionDef);
+
+                    //to do implement an ElmTypeInferencingVisitor; make sure that the resultType is set for each node
+                    if(!(expressionDef.getExpression() instanceof  SingletonFrom)) {
```

Comment:
This can just be a simple test against resultType. If any expressionDef has a null resultType, the ELM cannot be used in the translator.

Reply:
Done